iocore/net/UnixNetVConnection.cc

```diff
@@ -270,6 +270,21 @@ read_from_net(NetHandler *nh, UnixNetVConnection *vc, EThread *thread)
     close_UnixNetVConnection(vc, thread);
     return;
   }
+
+  if (vc->read.error) {
+    int err = 0, errlen = sizeof(int);
+    if (getsockopt(vc->con.fd, SOL_SOCKET, SO_ERROR, &err, (socklen_t *)&errlen) == -1) {
+      err = errno;
+    }
+    // if it is a non-temporary error, we should die appropriately
+    if (err != EAGAIN && err != EINTR) {
+      read_signal_error(nh, vc, err);
+      return;
+    } else {
+      vc->read.error = 0;
```

Comment:
@jpeach the error clear here. 

How to handle the EVENTIO_ERROR:
1. Set error to 1 if NetHandler get EVENTIO_ERROR on NetVC
2. Call net_read_io or write_to_net_io for the NetVC that has triggered and enabled or only has error).
3. In the io function, we get lock first then check the vio error status.
4. Try to get socket error by getsockopt if error set
5. clear vio error status if non-fatal error
6. call read|write_signal_error if fatal error and the NetVC should be closed.